packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/fetch_auth_session_state_machine.dart

```diff
@@ -263,7 +260,8 @@ class FetchAuthSessionStateMachine extends StateMachine<FetchAuthSessionEvent,
     emit(
       FetchAuthSessionState.success(
         CognitoAuthSession(
-          isSignedIn: userPoolTokens != null,
+          isSignedIn: userPoolTokens != null ||
+              result.signInDetails is CognitoSignInDetailsFederated,
```

Comment:
if `userPoolTokens == null && result.signInDetails is CognitoSignInDetailsFederated` then `isSignedIn=true`.
however line 284 checks for `if userPoolTokens == null` to set the results to signedOut. is this a wanted behaviour?

Reply:
It is correct, unfortunately. This is a very confusing API, but basically the way to think about it is that we have a single API which wraps Cognito user pools and identity pools. User pools actually have a user directory and manage users, whereas identity pools just vend AWS credentials and allow authenticating with external OIDC providers.

When calling `federateToIdentityPool`, you're doing the latter. You're taking a Google OIDC token and saying hey give me some AWS credentials - I am who I say I am. In this case, all the user management APIs in our library are shutdown since we cannot change user info on the Google profile.

The choice to have `isSignedIn=true` under federation is not something I agree with, but is the design for this API. That's what this line checks - `isSignedIn` means authenticated via a user pool or authenticated via a third-party in the identity pool.

The check on line 284 is a different requirement - that you cannot authenticate with _both_ a user pool and an identity pool. If you're already logged into a user pool, line 284 stops any attempt to federate into an identity pool.

Reply:
thanks for the detailed explanation. do we have test cases for cognito federated identity in `packages/auth/amplify_auth_cognito_test/test/state/fetch_auth_session_state_machine_test.dart`?

Reply:
Good catch! Meant to do this. Added 👍